test/framework/ownerreference_helpers.go

```diff
@@ -157,10 +157,6 @@ var ExpOwnerReferenceAssertions = map[string]func([]metav1.OwnerReference) error
 		// ClusterResourcesSet doesn't have ownerReferences (it is a clusterctl move-hierarchy root).
 		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{})
 	},
-	// ClusterResourcesSetBinding has both Cluster and ClusterResourceSet set as owners on creation.
```

Comment:
Only the clusterGVK please
```go
	// ClusterResourcesSetBinding has both Cluster and ClusterResourceSet set as owners on creation.
	clusterResourceSetBindingKind: func(owners []metav1.OwnerReference) error {
		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterResourceSetGVK})
	},
```
Otherwise we don't validate any ownerRefs for the ClusterResourceSetBinding